app/src/main/java/com/github/se/icebreakrr/ui/authentication/ProfileCreation.kt

```diff
@@ -0,0 +1,314 @@
+                            color =
+                                if (selectedGender == gender)
+                                    MaterialTheme.colorScheme.onPrimaryContainer
+                                else MaterialTheme.colorScheme.onSurface)
+                      }
+                }
+              }
+
+          TagSelector(
+              selectedTag = selectedTags.map { tag -> Pair(tag, tagsViewModel.tagToColor(tag)) },
+              outputTag =
+                  tagsSuggestions.value.map { tag -> Pair(tag, tagsViewModel.tagToColor(tag)) },
+              stringQuery = stringQuery.value,
+              expanded = expanded,
+              onTagClick = { tag ->
+                tagsViewModel.removeFilter(tag)
+                isModified = true
+              },
+              onStringChanged = {
+                tagsViewModel.setQuery(it, selectedTags)
+                isModified = true
+              },
+              textColor = MaterialTheme.colorScheme.onSurface,
+              onDropDownItemClicked = { tag ->
+                tagsViewModel.addFilter(tag)
+                isModified = true
+              },
+              height = screenHeight,
+              width = screenWidth,
+              textSize = (screenHeight.value * 0.03).sp)
+        }
+
+    // Add confirmation button in top right corner
+    IconButton(
+        onClick = {
+          if (isFormValid()) {
+            val currentUser = Firebase.auth.currentUser
+            currentUser?.let { user ->
+              FirebaseMessaging.getInstance().token.addOnCompleteListener { task ->
+                if (task.isSuccessful) {
+                  val fcmToken = task.result
+                  val newProfile =
+                      Profile(
+                          uid = user.uid,
+                          name = fullName.value,
+                          gender = selectedGender!!, // Add this line
+                          birthDate =
+                              Timestamp(
+                                  Calendar.getInstance()
+                                      .apply {
+                                        time = selectedDate.value!!
+                                        set(Calendar.HOUR_OF_DAY, 0)
+                                        set(Calendar.MINUTE, 0)
+                                        set(Calendar.SECOND, 0)
+                                        set(Calendar.MILLISECOND, 0)
+                                      }
+                                      .time),
+                          catchPhrase = catchphrase.value,
+                          description = description.value,
+                          tags = selectedTags,
+                          fcmToken = fcmToken)
+                  profilesViewModel.addNewProfile(newProfile)
+                  // Upload profile picture if selected
+                  profilesViewModel.validateAndUploadProfilePicture(context)
```

Comment:
Bug: to use the validateAndUploadProfilePicture you must set `selectedProfile` of `profilesViewModel` to the profile you just created, see why below:

![image](https://github.com/user-attachments/assets/a74b58c0-4c77-49d2-82df-272e54fe603d)